### ISSUE TITLE
Error 403 Forbidden in relay_server

### ISSUE THREAD

**Issue by Leen15** [sentry-kubernetes/charts#212]
Hello,
we installed sentry with helm chart, but we have an issue with the relay service.
When a client sends an event, the nginx service returns a 200 status code, but the event is not saved and in relay service this is the log:

```
2020-10-15T08:00:27Z [relay::setup] INFO: launching relay from config folder /work/.relay
2020-10-15T08:00:27Z [relay::setup] INFO:   relay mode: managed
2020-10-15T08:00:27Z [relay::setup] INFO:   relay id: be4c355d-473a-43ed-9a00-8c1a74cac646
2020-10-15T08:00:27Z [relay::setup] INFO:   public key: PyrBQl11uy3mhIQ4fzTZ2eR-lCkWae0ozmKu2FsU9v0
2020-10-15T08:00:27Z [relay::setup] INFO:   log level: INFO
2020-10-15T08:00:27Z [relay_server::actors::upstream] INFO: upstream relay started
2020-10-15T08:00:27Z [relay_server::actors::upstream] INFO: registering with upstream (http://sentry-web:9000/)
2020-10-15T08:00:27Z [relay_server::actors::outcome::processing] INFO: OutcomeProducer started.
2020-10-15T08:00:27Z [relay_server::actors::events] INFO: starting 16 event processing workers
2020-10-15T08:00:27Z [relay_server::actors::store] INFO: store forwarder started
2020-10-15T08:00:27Z [relay_server::actors::project_redis] INFO: redis project cache started
2020-10-15T08:00:27Z [relay_server::actors::project_redis] INFO: redis project cache started
2020-10-15T08:00:27Z [relay_server::actors::project_redis] INFO: redis project cache started
2020-10-15T08:00:27Z [relay_server::actors::project_redis] INFO: redis project cache started
2020-10-15T08:00:27Z [relay_server::actors::project_redis] INFO: redis project cache started
2020-10-15T08:00:27Z [relay_server::actors::project_redis] INFO: redis project cache started
2020-10-15T08:00:27Z [relay_server::actors::project_redis] INFO: redis project cache started
2020-10-15T08:00:27Z [relay_server::actors::project_redis] INFO: redis project cache started
2020-10-15T08:00:27Z [relay_server::actors::project_redis] INFO: redis project cache started
2020-10-15T08:00:27Z [relay_server::actors::project_redis] INFO: redis project cache started
2020-10-15T08:00:27Z [relay_server::actors::project_redis] INFO: redis project cache started
2020-10-15T08:00:27Z [relay_server::actors::project_redis] INFO: redis project cache started
2020-10-15T08:00:27Z [relay_server::actors::project_redis] INFO: redis project cache started
2020-10-15T08:00:27Z [relay_server::actors::project_redis] INFO: redis project cache started
2020-10-15T08:00:27Z [relay_server::actors::project_redis] INFO: redis project cache started
2020-10-15T08:00:27Z [relay_server::actors::project_redis] INFO: redis project cache started
2020-10-15T08:00:27Z [relay_server::service] INFO: spawning http server
2020-10-15T08:00:27Z [relay_server::service] INFO:   listening on: http://0.0.0.0:3000/
2020-10-15T08:00:27Z [actix_net::server::server] INFO: Starting 16 workers
2020-10-15T08:00:27Z [actix_net::server::server] INFO: Starting server on 0.0.0.0:3000
2020-10-15T08:00:27Z [relay_server::actors::controller] INFO: relay server starting
2020-10-15T08:00:27Z [relay_server::actors::connector] INFO: metered connector started
2020-10-15T08:00:27Z [relay_server::actors::events] INFO: event manager started
2020-10-15T08:00:27Z [relay_server::actors::project_local] INFO: project local cache started
2020-10-15T08:00:27Z [relay_server::actors::project_upstream] INFO: project upstream cache started
2020-10-15T08:00:27Z [relay_server::actors::project_cache] INFO: project cache started
2020-10-15T08:00:27Z [relay_server::actors::project_keys] INFO: project cache started
2020-10-15T08:00:27Z [relay_server::actors::relays] INFO: key cache started
2020-10-15T08:00:27Z [relay_server::actors::upstream] ERROR: authentication encountered error: could not send request to upstream
  caused by: Server disconnected
  caused by: Server disconnected
2020-10-15T08:00:27Z [relay_server::actors::upstream] INFO: registering with upstream (http://sentry-web:9000/)
2020-10-15T08:00:27Z [relay_server::actors::upstream] ERROR: authentication encountered error: could not send request to upstream
  caused by: Server disconnected
  caused by: Server disconnected
2020-10-15T08:00:28Z [relay_server::actors::upstream] INFO: registering with upstream (http://sentry-web:9000/)
2020-10-15T08:00:28Z [relay_server::actors::upstream] ERROR: authentication encountered error: could not send request to upstream
  caused by: Server disconnected
  caused by: Server disconnected
2020-10-15T08:00:30Z [relay_server::actors::upstream] INFO: registering with upstream (http://sentry-web:9000/)
2020-10-15T08:00:30Z [relay_server::actors::upstream] INFO: relay successfully registered with upstream
2020-10-15T08:38:00Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T08:38:01Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T08:38:02Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T08:38:05Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T08:38:08Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T08:38:13Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T08:38:21Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T08:38:33Z [relay_server::actors::events] ERROR: error processing event: failed to resolve project information
  caused by: failed to fetch project state from upstream
2020-10-15T10:29:38Z [r2d2] ERROR: Connection timed out (os error 110)
2020-10-15T10:29:38Z [r2d2] ERROR: Connection timed out (os error 110)
2020-10-15T10:29:38Z [r2d2] ERROR: Connection timed out (os error 110)
2020-10-15T12:30:51Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:30:52Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:30:54Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:30:56Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:30:59Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:31:05Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:31:12Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:31:24Z [relay_server::actors::events] ERROR: error processing event: failed to resolve project information
  caused by: failed to fetch project state from upstream
2020-10-15T12:31:24Z [relay_server::actors::events] ERROR: error processing event: failed to resolve project information
  caused by: failed to fetch project state from upstream
2020-10-15T12:31:24Z [relay_server::actors::events] ERROR: error processing event: failed to resolve project information
  caused by: failed to fetch project state from upstream
2020-10-15T12:38:47Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:38:48Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:38:49Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:38:52Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:38:55Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:39:01Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:39:08Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:39:20Z [relay_server::actors::events] ERROR: error processing event: failed to resolve project information
  caused by: failed to fetch project state from upstream
2020-10-15T12:41:51Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:41:52Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:41:54Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:41:56Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:42:00Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:42:05Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:42:13Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:42:24Z [relay_server::actors::events] ERROR: error processing event: failed to resolve project information
  caused by: failed to fetch project state from upstream
2020-10-15T12:46:11Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:46:12Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:46:14Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:46:16Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:46:20Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:46:25Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:46:33Z [relay_server::actors::project_upstream] ERROR: error fetching project states: upstream request returned error 403 Forbidden
  caused by: no error details
2020-10-15T12:46:44Z [relay_server::actors::events] ERROR: error processing event: failed to resolve project information
  caused by: failed to fetch project state from upstream
```

We tried to debug the reason, and we found a table `sentry_relay` in the postgreSQL database with a list or relays (id, public key etc).
Here, there is a column `is_internal` that is set as FALSE. If we manually change the value to TRUE, there is no more errors in relay service log and events are successfully processed.
We don't understand what is the scope of this flag and why it's false.

Somebody can help us?
Every time the relay service will be redeployed by kubernetes, network error or helm upgrade, we will lose the main functionality of sentry (logs events) and we cannot manually update this value every time.

Thanks

**Comment by arizz96** [comment]
I'm experiencing the same issue

**Comment by slv** [comment]
Same here

**Comment by Mokto** [comment]
Could you guys try the proxy mode of relay, please ?

**Comment by Leen15** [comment]
@Mokto How can I do it? 
I didn't find any information in values file about it.

**Comment by Mokto** [comment]
https://github.com/sentry-kubernetes/charts/blob/develop/sentry/values.yaml#L38

mode: proxy

**Comment by Leen15** [comment]
@Mokto it works!

**Comment by Mokto** [comment]
Fixed through 7.0.0.
Thanks for reporting!
